internal/controller/supervisorconfig/githubupstreamwatcher/github_upstream_watcher.go

```diff
@@ -20,25 +21,27 @@ import (
 	idpinformers "go.pinniped.dev/generated/latest/client/supervisor/informers/externalversions/idp/v1alpha1"
 	pinnipedcontroller "go.pinniped.dev/internal/controller"
 	"go.pinniped.dev/internal/controller/conditionsutil"
+	"go.pinniped.dev/internal/controller/supervisorconfig/upstreamwatchers"
 	"go.pinniped.dev/internal/controllerlib"
 	"go.pinniped.dev/internal/federationdomain/upstreamprovider"
 	"go.pinniped.dev/internal/plog"
 	"go.pinniped.dev/internal/upstreamgithub"
 )
 
 const (
-	// Setup for the name of our controller in logs.
 	controllerName = "github-upstream-observer"
 
 	// Constants related to the client credentials Secret.
-	gitHubClientSecretType corev1.SecretType = "secrets.pinniped.dev/github-client"
-
-	//
-	// clientIDDataKey     = "clientID"
-	// clientSecretDataKey = "clientSecret"
-	//
-	// // Constants related to conditions.
-	// typeClientCredentialsValid = "ClientCredentialsValid" //nolint:gosec // this is not a credential.
+	gitHubClientSecretType               corev1.SecretType = "secrets.pinniped.dev/github-client"
+	clientIDDataKey, clientSecretDataKey string            = "clientID", "clientSecret"
+
+	HostValid                string = "HostValid"
+	TLSConfigurationValid    string = "TLSConfigurationValid"
+	OrganizationsPolicyValid string = "OrganizationsPolicyValid"
+	// ClientCredentialsObtained is different from other status conditions because it only checks that the client credentials
+	// have been obtained. The controller has no way to verify whether they are valid.
+	ClientCredentialsObtained string = "ClientCredentialsObtained"
```

Comment:
Looking at a couple others:

- [UpstreamWatchers Type & Reason](https://github.com/vmware-tanzu/pinniped/blob/722f00e485abbb6f0a2ca4b616feb1375c5ae783/internal/controller/supervisorconfig/upstreamwatchers/upstream_watchers.go#L37)
- [OIDCClientValidatior Type & Reason](https://github.com/vmware-tanzu/pinniped/blob/722f00e485abbb6f0a2ca4b616feb1375c5ae783/internal/federationdomain/oidcclientvalidator/oidcclientvalidator.go#L22)
- [JWT Authenticator Type & Reason & Message](https://github.com/vmware-tanzu/pinniped/blob/main/internal/controller/authenticator/jwtcachefiller/jwtcachefiller.go#L48)
- [WebHook AUthenticator Type & Reason & Message](https://github.com/vmware-tanzu/pinniped/blob/main/internal/controller/authenticator/webhookcachefiller/webhookcachefiller.go#L44)

Usually `Valid`, but not always, so this deviation may make sense.